### ISSUE TITLE
[Feature Request] Barcode generation for Licenses, Accessories, Consumables, and Components

### ISSUE THREAD

**Issue by 08solsticegxp** [snipe/snipe-it#3957]
#### Expected Behavior (or desired behavior if a feature request)

Have the ability to generate a barcode for Licenses, Accessories, Consumables, and Components just like you can for assets.

The desire is to be able to use this for storage room inventory. I would like to be able to put my keyboards, mice, phones, laptops, ect... on storage shelves and have a barcode (2D QRCode) that I can scan and will take me right to the page where I can check out the item (even better would be just remove one from the qty and have the option to not have to assign it to someone)

Right now you can only do this with Asset items, but you can't say you have 30 of the same item.

I did try to use a QRcode generator with a URL such as 
> http://snipeit/admin/accessories/1/view
But that just ended up taking me to a login page and then after logging in, to the dashboard.
-----

#### Actual Behavior

There appears to be no barcode generation available for anything other than assets.

-----

#### Please confirm you have done the following before posting your bug report:

- [ ] I have enabled debug mode
- [x] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

-----
#### Please provide answers to these questions before posting your bug report:

- Version of Snipe-IT you're running: Version v3.6.6 build 1 (gee2c67a)
- What OS and web server you're running Snipe-IT on: Centos 6.9 Apache
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc) Manual installation. Install.sh appears broken.
- WITH DEBUG TURNED ON, if you're getting an error in your browser, include that error: no Errors
- What specific Snipe-IT page you're on, and what specific element you're interacting with to trigger the error
- If a stacktrace is provided in the error, include that too.
- Any errors that appear in your browser's error console.
- Confirm whether the error is [reproduceable on the demo](https://snipeitapp.com/demo).
- Include any additional information you can find in `app/storage/logs` and your webserver's logs.
- Include what you've done so far in the installation, and if you got any error messages along the way: only got errors with the install.sh saying it couldn't complete the install because of the OS version. I just restarted and manually installed. There were no errors when manually installing.
- Indicate whether or not you've manually edited any data directly in the database: no custom editing.

**Comment by ideaman24** [comment]
I agree with this thought. I have different components that I would like to tag with a label. However, this would also entail adding a asset or component ID to the mix like assets have.

**Comment by mlawday** [comment]
+1

**Comment by Chroniclehome** [comment]
I agree, with this. We use the program, which we love btw, and the ability to make labels for the components specifically would make life, and inventory counts, so much easier. Thanks for considering it!

**Comment by rpesta** [comment]
+1 to re-open

**Comment by anonimust** [comment]
Yes will need this feature as the business I am working for have consumables and need a way to scan labels to check-in/out inventory from scanning and to keep up with the inventory automatically. Kinda need this as soon as possible as they are currently looking for different inventory management services and this one would be perfect if this feature was enabled.

**Comment by snipe** [comment]
@anonimust This won't be implemented for at *least* another month or two at the earliest.

**Comment by osa-ko** [comment]
This is indeed a critical feature. We have to log pretty much all the accessories now as assets as we need them to have a barcode label attached. Please implement this.

**Comment by enoughsev** [comment]
It would be great to have a generic barcode for each item and everytime you scan this barcode we can lower the item or increase the item. We have hundreds of different accessories we keep on hand and having to search for each is time-consuming.

**Comment by ITHelper2010** [comment]
Any update to this request. Very interested and need as well.

**Comment by maelask3** [comment]
This is the feature that is a deal-breaker for me in terms of adopting Snipe-IT, as I do need to keep track of a lot of different accessories, components, etc.

**Comment by giltee** [comment]
This would be a really amazing feature to add to the application. I can see Snipe indicated that this was at least a couple months out (in the beginning of August). Can anyone confirm if this is actively being worked on?

**Comment by viclou** [comment]
FD#8691 +1

**Comment by C14VCO** [comment]
+ 1

**Comment by rpesta** [comment]
Still relevant bot.

**Comment by avm5689** [comment]
+1 for this!

**Comment by cliff-pennalligen** [comment]
+1, Snipe IT is fantastic, BTW :)

**Comment by danysanblas1** [comment]
+1 for this please

**Comment by danysanblas1** [comment]
> @anonimust This won't be implemented for at _least_ another month or two at the earliest.

.

**Comment by giltee** [comment]
Two months is not that long, this is still relevant for sure.

**Comment by wallacj** [comment]
yes please this is very needed ....    Love snipit

**Comment by Jackano** [comment]
We are now using Snipe-It and our final user requested it in the first two weeks. Extend the barcode generation to every objects and not only assets. Must have/must add!

**Comment by TJucutan** [comment]
Like others have mentioned, being able to have a single barcode for consumables that could be scanned to add or deduct. Maybe instead of having the consumable screen being similar to the Asset list, it can be more like the check-in/checkout screen, and once you scan a barcode it will ask if you want to add or subtract and you can input the quantity if working with more than 1. 

In my use case, we have boxes for shipping logged in consumables and it would be nice to just scan once and say how many boxes were used for a particular shipment.

**Comment by talpait** [comment]
+1 for this!

**Comment by Jackano** [comment]
Any news on this please?

**Comment by aleksandrmetik** [comment]
+1

**Comment by hxcsp** [comment]
+1

We just implemented Snipe-IT and it seems to be lacking this key feature.  This would be a great addition. 

Thanks!

**Comment by chndler** [comment]
+1

We're rolling out Snipe now and this would definitely be appreciated. I'd be happy to try implementing this feature given some guidance.

**Comment by pixeldj** [comment]
+1

Would love if this was a part of Snipe-IT

**Comment by joel-almoosa** [comment]
+1 i just adapted SnipeIT to handle our company assets and first thing that I wanted was this feature. We need to have accessories and components to have tags so we can also uniquely monitor them on our asset lists.

**Comment by 08solsticegxp** [comment]
Request was made 2 years ago. I doubt it's coming. Quite disappointed.

**Comment by ajansamel** [comment]
+1 This would be ideal feature to add.

**Comment by cliff-pennalligen** [comment]
> Request was made 2 years ago. I doubt it's coming. Quite disappointed.
08solsticegxp - respectfully, I acknowledge your frustration.  But like everyone, the creators of Snipe IT also need to eat !! :)  Do you pay (e.g. through hosting or sponsorship)?  The more funding they have, the more they can work on features like this.

**Comment by pbrink231** [comment]
+1,  running into this issue also.

**Comment by kendzioc** [comment]
Would also love to have barcodes for locations.

**Comment by rem9000** [comment]
+1

**Comment by tareko** [comment]
Not sure how else to add my voice, but this feature would help organize things well. As a hack, we're putting our consumables as assets, which is a bit more messy.

**Comment by ReyoMediaCY** [comment]
+1

**Comment by ced455** [comment]
+1

**Comment by litlebooks** [comment]
Yes!!! This feature is so necessary!

**Comment by stuartm** [comment]
Really surprised to find this isn't already possible. Components etc still need to be tracked, maybe even more so than larger less portable items like assets. Knowing that the component that was checked out was the one that was returned, or where a found item properly belongs is critical to proper inventory management. Right now it seems the only option it to have everything as an asset which is itself not ideal because you then lose the relationship tracking of which component is associated with an asset.

**Comment by petersont4** [comment]
I use a QR Generator and the url https://develop.snipeitapp.com/accessories/1 to create these.  I know it isn't a built in solution but it works for me for the time being.   OP said that it brings them to the login screen then to the dashboard.  Mine brings me to the login screen if I am not already logged in, then to the page the QR code points at.


<blockquote><img src="https://develop.snipeitapp.com/favicon.ico" width="48" align="right"><div><strong><a href="https://develop.snipeitapp.com/accessories/1">Snipe-IT Asset Management Demo</a></strong></div></blockquote>

**Comment by samuelT2** [comment]
+1

**Comment by BeanHubbleday** [comment]
Yep +1 for this one!

**Comment by TinaMccray** [comment]
+1

**Comment by mikedmor** [comment]
+1

**Comment by jamie-oconnell** [comment]
+1

**Comment by derrynj** [comment]
+1

**Comment by printful-dbdiagram** [comment]
+1

**Comment by focusmade** [comment]
Let's get this feature rolling! Four years of people asking is more than enough proof of interest.

**Comment by aleksandrmetik** [comment]
This is the obvious thing. I left my interest here over a year ago and I see that so far it has not moved on.
@snipe help us, please
Thanks

**Comment by danysanblas1** [comment]
I recomend to lean some programming languaje and make it by yourselft becouse it is a free versión that never gonna fo wharever you ask

Get Outlook para Android<https://aka.ms/AAb9ysg>

________________________________

**Comment by samuelT2** [comment]
@danysanblas1 dude what?

**Comment by musover** [comment]
> I recomend to lean some programming languaje and make it by yourselft becouse it is a free versión that never gonna fo wharever you ask

I already have learned some programming languages myself, however I do not have the PHP expertise needed to complete this task.
Also, a free sofware project is not bound to "never implement what users ask".

Your comment is simply uncalled for in this context.

**Comment by stuartm** [comment]
> I recomend to lean some programming languaje and make it by yourselft becouse it is a free versión that never gonna fo wharever you ask

https://github.com/snipe/snipe-it#bug-reports--feature-requests

SnipeIT invite feature requests. This is a feature request. 


<blockquote><img src="https://repository-images.githubusercontent.com/14346979/164a6580-7d79-11e9-865f-eae87593e11a" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/snipe/snipe-it">snipe/snipe-it</a></strong></div><div>A free open source IT asset/license management system - snipe/snipe-it</div></blockquote>

**Comment by modestos** [comment]
+1 must have feature

**Comment by Calog18** [comment]
I'd love to be able to scan a label for consumables like tape, cans of product etc that we use. So I know when the level drops to reorder. +1 too

**Comment by Zythyx** [comment]
Adding my support for this feature

**Comment by ScottAMains** [comment]
This would be great to have. Either a data matrix or QR for simple check-in/ check-out of accessories/components associated with assets and larger assemblies as a whole.

**Comment by gt2416** [comment]
Yup would love to see this implemented. Otherwise great product !

**Comment by aleksandrmetik** [comment]
There are things that are permanent. It's how water flows, how fire burns, how people ask to implement functionality on github for 4 years...

**Comment by Nicalicious** [comment]
This is the only feature that is missing that is keeping me from jumping in with both feet. It would also be nice if some of the major Manufacturers were built in to the installation. Seems silly for most everyone to have to manually add Dell, HP, Cisco,  Microsoft, Apple, and so on.

**Comment by Tylermac2** [comment]
Snipe-IT is an great tool, but this is a must feature to take it to the next level. Hopefully will see it soon in a V6 release. 
+1 Here!

**Comment by snipe** [comment]
@aleksandrmetik you missed one: people asking for features who don’t actually contribute. This, too, is like water.

It’s on the list. Lots of things are. Comments like this make it sound like we’re not listening and not moving the needle, which demonstrably untrue - and frankly kinda rude. 

We bust our asses every single day to provide a product that meets the needs of as many people as we can. You only care about your issues, but we have to care about the issues of tens of thousands of users. You just sound like an entitled prick right now. 

I welcome your pull request if you think you can do better.

**Comment by aleksandrmetik** [comment]
Hello @snipe 
I left three comments here:
https://github.com/snipe/snipe-it/issues/3957#issuecomment-567498784 - Dec 19, 2019 
"+1"

https://github.com/snipe/snipe-it/issues/3957#issuecomment-840806398 - May 13, 2021
"This is the obvious thing. I left my interest here over a year ago and I see that so far it has not moved on.
@snipe help us, please
Thanks."

https://github.com/snipe/snipe-it/issues/3957#issuecomment-954500462 - October 29, 2021
"There are things that are permanent. It's how water flows, how fire burns, how people ask to implement functionality on github for 4 years..."

I just wanted to point out that this issue is important, at least for me and >50 people here. I did not insult you or minimize  the importance of your work.
**I'm not quite sure where I went to personal insults for you to have the right to call me an entitled prick right now**.
**Cool down.** This is feedback that this ticket is important to some of your users. If you don't need it, just say so and close the ticket
Thanks

**Comment by snipe** [comment]
"This is the obvious thing. I left my interest here over a year ago and I see that so far it has not moved on."

That's a thing you said, and you expect us to prioritize your concerns? You flat out *told us* you gave up. 

I didn't call you an entitled prick, I said you *sound* like an entitled prick - because you do. Not everybody gets what they want when they want it. It's unfortunate, and nobody is more bummed by that than me, but it's also true. You not understanding how prioritization works in a large project is not a personal insult, and your words have consequences. If you want to act like a brat, be prepared to be treated like one. 

This is open source software with over a thousand requests open at any given time. We don't just prioritize based on how long an issue has been open. 

Otherwise your options are to be patient, put together a PR, or find something else you get for free that meets your needs. Nobody is holding you here. Bitching on GitHub does not provide any solutions to anyone. 

![pull-requests-welcome](https://user-images.githubusercontent.com/197404/152226797-3433624f-50b9-4d6f-81d6-f3102ed53552.jpg)

**Comment by aleksandrmetik** [comment]
@snipe 
You perceive information exclusively in a negative way. Even the joke about water and the crowd begging in the github. Unfortunately, I do not see any perspective to continue communicating with you.  
As for your claims, I'm not a PHP developer, I'm a DevOps engineer. But after talking to you, I have no desire to offer any help or donations.  It's a shame that such a good product is being run by you as a CEO.
Have a nice day.

**Comment by giltee** [comment]
@snipe as a leader of developers with way more work than we could possibly
ever achieve , I do understand where you’re coming from 100%. But I would
suggest you get one of your employees to comment on the forums going
forward. The people here just want to solve a problem. they may lack
understanding on development and the work associated with building a
feature. But, they certainly don’t deserve to be called names and berated.
If these small things bother you that much, it might be time for a
vacation. Otherwise keep your chin up and continue the good work.

**Comment by snipe** [comment]
@giltee I did not call anyone names nor was I berating. Words have consequences and it is inappropriate for someone to bully someone else because they haven't implemented a feature you wanted.

It's not that small things bother me - it's that they keep happening. People continue to treat open source projects worse than they treat the projects they pay for.  This has been happening forever, and I'm not going to sugar-coat it. I never have, and I never will. We bust our asses, and the response we often get is sarcastic and passive-aggressive commentary that adds literally zero to the discourse and does not provide solutions. 

If someone comes in correct, they will be treated correctly. This has been true for the nine years I've been running this project. I will not outsource this to an employee.

**Comment by snipe** [comment]
I'm locking this thread. There is no additional useful information we can get to here. We know what the ask is, and it's on a very long list of features we will work towards once we get v6 out. 

(Edited for typo-fix)

**Comment by uberbrady** [comment]
I don't understand why "please be nice to the developers who give this software away for free or they might snap at you" is such a controversial take.

It's so weird that our paying customers who ask for things (and to whom we say "no" all the time) are nicer to us than the people here.

Be nice. That's all we ask. Don't be mean. Ask for what you want, throw in your two cents wherever you need. But be nice.

And if you're not, we might get snippy. Doesn't that seem pretty fair? That's literally how humans work.
